lib/resources/mount.rb

```diff
@@ -49,12 +49,38 @@ def method_missing(name)
       line = mounted.stdout.lines.to_a.last if mounted.stdout.lines.count > 1
 
       # parse content if we are on linux
-      @mount_options ||= parse_mount_options(line)
+      @mount_options ||= @mount_manager.parse_mount_options(line)
       @mount_options[name]
     end
 
     def to_s
       "Mount #{@path}"
     end
+
+    private
+
+    def mount_manager_for_os
+      os = inspec.os
+      if os.linux?
+        LinuxMounts.new(inspec)
+      elsif ['freebsd'].include?(os[:family])
+        BsdMounts.new(inspec)
+      end
+    end
+  end
+
+  class MountsInfo
+    attr_reader :inspec
+    def initialize(inspec)
+      @inspec = inspec
+    end
+  end
```

Comment:
Should we add a default `#parse_mount_options` method here that `raise`s an exception to make it clear for future users that the method is required?